chapter_recurrent-neural-networks/lang-model.md

```diff
@@ -60,22 +60,22 @@ Here the coefficients $\epsilon_i > 0$ determine how much we use the
 estimate for a shorter sequence as a fill-in for longer
 ones. Moreover, $m$ is the total number of words we encounter. The
 above is a rather primitive variant of what is Kneser-Ney smoothing
-and Bayesian Nonparametrics can accomplish. See e.g., the Sequence
-Memoizer of Wood et al., 2012 for more details of how to accomplish
-this. Unfortunately, models like this get unwieldy rather quickly:
-first off, we need to store all counts and secondly, this entirely
-ignores the meaning of the words. For instance, *'cat'* and *'feline'*
-should occur in related contexts. Deep learning based language models
-are well suited to take this into account. This, it is quite difficult
-to adjust such models to additional context. Lastly, long word
+and Bayesian Nonparametrics can accomplish. See e.g., :cite:`Wood.Gasthaus.Archambeau.ea.2011` for more details of how to accomplish
+this. Unfortunately, models like this get unwieldy rather quickly 
+for the following reasons. First, we need to store all counts.
+Secondly, this entirely ignores the meaning of the words. For 
+instance, *'cat'* and *'feline'* should occur in related contexts. 
+It is quite difficult to adjust such models to additional context, 
+whereas, deep learning based language models are well suited to 
+take this into account.  Lastly, long word
 sequences are almost certain to be novel, hence a model that simply
 counts the frequency of previously seen word sequences is bound to
 perform poorly there.
 
 
 ## Markov Models and $n$-grams
 
-Before we discuss solutions involving deep learning we need some more terminology and concepts. Recall our discussion of Markov Models in the previous section. Let us apply this to language modeling. A distribution over sequences satisfies the Markov property of first order if $p(w_{t+1}|w_t, \ldots, w_1) = p(w_{t+1}|w_t)$. Higher orders correspond to longer dependencies. This leads to a number of approximations that we could apply to model a sequence:
+Before we discuss solutions involving deep learning, we need some more terminology and concepts. Recall our discussion of Markov Models in the previous section, let us apply this to language modeling. A distribution over sequences satisfies the Markov property of first order if $p(w_{t+1}|w_t, \ldots, w_1) = p(w_{t+1}|w_t)$. Higher orders correspond to longer dependencies. This leads to a number of approximations that we could apply to model a sequence:
```

Comment:
Recall something. Let us do this.
OR
Recalling something, let us do this.

Reply:
Sure